sources/api/netdog/src/main.rs

```diff
@@ -10,23 +10,38 @@ It contains two subcommands meant for use as settings generators:
 * `generate-hostname`: returns the node's hostname in JSON format. If the lookup is unsuccessful, the IP of the node is used.
 
 The subcommand `set-hostname` sets the hostname for the system.
+
+The subcommand `generate-net-config` generates the network interface configuration for the host. If
+a `net.toml` file exists in `/var/lib/bottlerocket`, it is used to generate the configuration. If
+`net.toml` doesn't exist, the kernel command line `/proc/cmdline` is checked for the prefix
+`netdog.default-interface`.  If an interface is defined with that prefix, it is used to generate an
+interface configuration.  A single default interface may be defined on the kernel command line with
+the format: `netdog.default-interface=interface name:option1,option2`.  "interface name" is the
+name of the interface, and valid options are "dhcp4" and "dhcp6".  A "?" may be added to the option
```

Comment:
nit: to avoid ambiguity about whether spaces are allowed:
```suggestion
the format: `netdog.default-interface=interface-name:option1,option2`.  "interface-name" is the
name of the interface, and valid options are "dhcp4" and "dhcp6".  A "?" may be added to the option
```